### PR TITLE
Worker load can be supplied by the game

### PR DESCRIPTION
**Pull request by improbable-danny**
#### Description
Adding a method to allow worker load to be supplied by the game.

#### Tests
Tested using the test gyms project

STRONGLY SUGGESTED: How can this be verified by QA?

#### Documentation
We should keep this as an internal only feature

#### Primary reviewers
@m-samiec

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp [hunk 1]
```diff
@@ -23,6 +23,8 @@ void USpatialMetrics::Init(USpatialWorkerConnection* InConnection, float InNetSe
 	FramesSinceLastReport = 0;
 	TimeOfLastReport = 0.0f;
 
+	WorkerLoad = 0.0;
```
Comment (MatthewSandfordImprobable) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp:26]: Nit: 0.0f

Comment (improbable-danny) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp:26]: WorkerLoad is a double here!

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp [hunk 2]
```diff
@@ -23,8 +23,12 @@ void USpatialMetrics::Init(USpatialWorkerConnection* InConnection, float InNetSe
...
```
Comment (MatthewSandfordImprobable) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp:26]: WorkerLoad = 0.0f;

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp [hunk 3]
```diff
@@ -23,8 +23,12 @@ void USpatialMetrics::Init(USpatialWorkerConnection* InConnection, float InNetSe
...
+
 	bRPCTrackingEnabled = false;
 	RPCTrackingStartTime = 0.0f;
+
+	WorkerLoadDelegate.BindUObject(this, &USpatialMetrics::CalculateLoad);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp:31]: Can remove this

File: CHANGELOG.md [hunk 4]
```diff
@@ -80,6 +80,7 @@ Usage: `DeploymentLauncher createsim <project-name> <assembly-name> <target-depl
 - Add ability to disable outgoing RPC queue timeouts by setting `QueuedOutgoingRPCWaitTime` to 0.0f.
 - Added `bWorkerFlushAfterOutgoingNetworkOp` (defaulted false) which publishes changes to the GDK worker queue after RPCs and property replication to allow for lower latencies. Can be used in conjunction with `bRunSpatialWorkerConnectionOnGameThread` to get the lowest available latency at a trade-off with bandwidth.
 - You can now edit the project name field in the `Cloud Deployment` window.
+- Worker load can be specified by game logic via `SpatialMetrics::SetWorkerLoadFunction`
```
Comment (m-samiec) [inline CHANGELOG.md:83]:
```suggestion
- Worker load can be specified by game logic via `SpatialMetrics::SetWorkerLoadDelegate`
```

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp [hunk 5]
```diff
@@ -56,7 +67,7 @@ void USpatialMetrics::TickMetrics(float NetDriverTime)
 	Connection->SendMetrics(DynamicFPSMetrics);
 }
 
-// Load defined as performance relative to target frame time or just frame time based on config value.
+// Load defined as performance relative to target frame time or just frame time based on config value
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp:68]:
```suggestion
// Load defined as performance relative to target frame time or just frame time based on config value.
```

**Pull request merged**